### DISCUSSION TITLE
Rotor effective turbulence

### DISCUSSION THREAD

**Discussion by MYMahfouz** [OpenFAST/openfast#2107] [Q&A] [answered]
Dear all,

Can I use the output channel `RtVAvgxh` to calculate the rotor effective turbulence?
I know that this channel is affected by the tower top motions, and hence carries other frequencies from the structure.

What are the other channels I can use?

Best regards,
Youssef

**Comment by jjonkman** [comment]
Dear @MYMahfouz,

By “effective wind speed”, presumably you are referring to the average ambient wind speed across the rotor disk; is that correct?

AeroDyn’s output `RtVAvgxh` is the relative wind speed at each aerodynamic analysis node, averaged across the rotor and projected along the shaft axis, which is probably the closest output to what you want, but as you said, includes structural motion effects.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman,

Thank you for your reply. Sorry I think my question was not clear.

I want to get the average ambient turbulence on the rotor inside FAST.Farm by calculating the standard deviation of wind speed divided by the mean value. 

So which output channel can I use to get the turbulence acting on the rotor?

I am thinking one of the following approaches:
- Using `WindDiskVelX` from inflowwind output.
- Using `RtVAvgxh` from aerodyn.
- Creating a 9 points grid and extracting the values from inflowwind and then use these values to calculate the average turbulence.

Which of these approaches is better?
Is there a channel where I can directly get the turbulence value in FAST.Farm?

Best regards,
Youssef

**Reply by jjonkman** [reply, marked as answer]
Dear @MYMahfouz,

I would think the latter would be the best.  If you use the `WindDiskVelX` output from InflowWind or the `RtVAvgxh` output from AeroDyn, these outputs already include spatial averaging, and so, the turbulence calculation would be smoothed out.

FAST.Farm output `TIAmbTα` for turbine α is the ambient turbulence intensity calculated by FAST.Farm the given rotor, as is used in the eddy viscosity model, but I suspect this is not the turbulence output you want.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman,

I have a final question just to be sure I am setting the grid inside inflowwind properly.

What is the reference frame used to define the grid in Inflowwind? Is it a fixed reference frame or is it relevant to the rotor hub? 
So if the turbine sways would this mean that some of the points defined are no longer on the rotor?

Best regards,

**Reply by jjonkman** [reply]
Dear @MYMahfouz.

The reference frame used by InflowWind is an inertial frame; it does not move with floater or turbine motions.  This is why you must always define the wind domain to be larger than the dimensions of the turbine, so, that the turbine remains within the wind domain even when experiencing displacements/deflections.

Best regards

**Comment by mohsamaz** [comment]
Dear @jjonkman,
 I have a question regarding the calculation of WindDiskVelX. Is it defined as the spatially averaged wind velocity in the x-direction?

Specifically, I'm trying to determine whether, in a TurbSim-generated turbulent wind field, the average wind speed over a circular region(radius of 63m for 5MW turbine) at the center of each plane (for every time step) would be equal to WindDiskVelX.

Thank you for your help!

Best regards,

**Reply by jjonkman** [reply]
Dear @mohsamaz,

In InflowWind, output parameter `WindDiskVelX `is computed by spatially averaging the ambient wind velocities in the rotor plane around an annular sector at a radius at 70% span and projecting that annular average along the X axis of the inertial frame.

Best regards,

**Reply by yashpatil-pzwind** [reply]
Dear @jjonkman,
So is it 70% from the tip of blade or from center? Also, Does it take average from observing whole grid or only 9 nodes where blade nodes are supposed to be? If I were to read this logic where can I check? I tried checking Inflowind Module from OpenFAST Repository I didnt get much clarity.

Correct me if I am wrong.

Best Regards,

**Reply by jjonkman** [reply]
Dear @yashpatil-pzwind,

Yes, it is 70% of the rotor radius from the hub center, averaging across the whole annulus.  See the calculation in `SUBROUTINE` `InflowWind_GetRotorSpatialAverage()` of _InflowWind_Subs.f90_.

Best regards,
